eng/common/scripts/Package-Properties.ps1

```diff
@@ -76,81 +76,76 @@ function Get-PkgProperties
 {
     Param
     (
-        [Parameter(Mandatory = $true)]
         [string]$PackageName,
-        [Parameter(Mandatory = $true)]
+        [string]$ArtifactName,
         [string]$ServiceDirectory
     )
 
-    $pkgDirectoryPath = $null
-    $serviceDirectoryPath = Join-Path $RepoRoot "sdk" $ServiceDirectory
-    if (!(Test-Path $serviceDirectoryPath))
+    if (!$PackageName -and !$ArtifactName)
     {
-        LogError "Service Directory $ServiceDirectory does not exist"
+        LogError "You must specify either PackageName or ArtifactName Argument"
         return $null
     }
-
-    $directoriesPresent = Get-ChildItem $serviceDirectoryPath -Directory
-
-    foreach ($directory in $directoriesPresent)
+    
+    if ($ServiceDirectory)
     {
-        $pkgDirectoryPath = Join-Path $serviceDirectoryPath $directory.Name
-
-        if ($GetPackageInfoFromRepoFn -and (Test-Path "Function:$GetPackageInfoFromRepoFn"))
+        if (Test-Path $ServiceDirectory)
         {
-            $pkgProps = &$GetPackageInfoFromRepoFn -pkgPath $pkgDirectoryPath -serviceDirectory $ServiceDirectory -pkgName $PackageName
+            $ServiceDirectoryName = Split-Path $ServiceDirectory -Leaf
         }
-        else
+        else 
         {
-            LogError "The function for '$GetPackageInfoFromRepoFn' was not found.`
-            Make sure it is present in eng/scripts/Language-Settings.ps1 and referenced in eng/common/scripts/common.ps1.`
-            See https://github.com/Azure/azure-sdk-tools/blob/master/doc/common/common_engsys.md#code-structure"
+            $ServiceDirectoryName = $ServiceDirectory
         }
+        $AllPkgProps = Get-AllPkgProperties -ServiceDirectoryName $ServiceDirectoryName
+    }
+    else 
+    {
+        $AllPkgProps = Get-AllPkgProperties
+    }
 
-        if ($pkgProps -ne $null)
+    foreach ($pkgProp in $AllPkgProps)
+    {
+        if($PackageName -and ($pkgProp.Name -eq $PackageName))
+        {
+            return $pkgProp
+        }
+
+        if ($ArtifactName -and ($pkgProp.ArtifactName -eq $ArtifactName))
         {
-            return $pkgProps
+            return $pkgProp
         }
     }
-    LogWarning "Failed to retrive Properties for $PackageName"
+
+    if ($PackageName) { LogWarning "Failed to retrive Properties for [ $PackageName ]" }
+    if ($ArtifactName) { LogWarning "Failed to retrive Properties for [ $ArtifactName ]" }
     return $null
 }
 
 # Takes ServiceName and Repo Root Directory
 # Returns important properties for each package in the specified service, or entire repo if the serviceName is not specified
 # Returns an Table of service key to array values of PS Object with properties @ { pkgName, pkgVersion, pkgDirectoryPath, pkgReadMePath, pkgChangeLogPath }
-function Get-AllPkgProperties ([string]$ServiceDirectory = $null)
+function Get-AllPkgProperties ([string]$ServiceDirectoryName = $null)
```

Comment:
I'm not sure it is worth renaming this parameter.

Reply:
I just want to make it less confusing.

Reply:
I believe in almost all other places we just call this ServiceDirectory so I think sticking with that name should be fine.